modules/fbs/fbs.module

```diff
@@ -520,3 +623,22 @@ function _fbs_wayf_login($name, $value = NULL) {
 
   return NULL;
 }
+
+/**
+ * Implements hook_ting_pre_execute().
+ *
+ * Filter out blacklisted branches from search.
+ */
+function fbs_ting_pre_execute($request) {
+  // We only want to modify searches.
+  if (gettype($request) != 'TingClientSearchRequest') {
+    return;
+  }
+  $fbs_branches_blacklist = variable_get('fbs_branches_blacklist', array());
+  $query = $request->getQuery();
+  if (!empty($fbs_branches_blacklist)) {
+    $query .= ' AND (holdingsitem.branch = (* not ' . implode(' not ',
+        $fbs_branches_blacklist) . '))';
+    $request->setQuery($query);
+  }
+}
```

Comment:
And elements in `$fbs_branches_blacklist` should be quoted, as `holdingsitem.branch` apparently is the string (ref: http://platform.dandigbib.org/issues/1618#note-31)

Which also means that `$fbs_branches_blacklist` cannot be used directly, but needs to be mapped from IDs to names. Alternatively, `$fbs_branches_blacklist` is an ID => name array, and this uses array_values() while the other uses array_keys().